### PR TITLE
refactor: remove with-stub-server macro

### PR DESCRIPTION
**Pull request by opqdonut**
prefer the run-with-server function

### CODE REVIEW

File: test/clj/rems/db/test_entitlements.clj [hunk 1]
```diff
@@ -127,131 +118,143 @@
...
+                                :actor member
+                                :accepted-licenses [lic-id1 lic-id2]}) ; now accept all licenses
+
+           (entitlements/process-outbox!)
+
+           (testing "db"
+             (is (= #{[applicant "resource1"] [applicant "resource2"]
+                      [member "resource1"] [member "resource2"]}
+                    (set (map (juxt :userid :resid) (db/get-entitlements {:application app-id}))))))
+           (testing "POST"
```
Comment (foxlynx) [inline test/clj/rems/db/test_entitlements.clj:186]: Maybe same actually with these POSTs, bit ambiquous cause it does not actually POST but checks that the action produced correct POST commands. I did not think of this last time.

File: test/clj/rems/db/test_entitlements.clj [hunk 2]
```diff
@@ -28,10 +28,12 @@
 
 (defn run-with-server
```
[the remark this answers is not in the document]
Comment (opqdonut) [reply test/clj/rems/db/test_entitlements.clj:29]: the server is used here and duplicated in test-end-to-end, I think that's fine for now

File: test/clj/rems/db/test_entitlements.clj [hunk 3]
```diff
@@ -127,131 +118,143 @@
...
+                                  :accepted-licenses [lic-id1]}) ; only accept some licenses
+             (is (= {applicant #{lic-id1 lic-id2}
+                     member #{lic-id1}}
+                    (:application/accepted-licenses (applications/get-application app-id))))
+
+             (entitlements/process-outbox!)
+
+             (testing "db"
+               (is (= #{[applicant "resource1"] [applicant "resource2"]}
+                      (set (map (juxt :userid :resid) (db/get-entitlements {:application app-id}))))))
```
[the remark this answers is not in the document]
Comment (opqdonut) [reply test/clj/rems/db/test_entitlements.clj:161]: it was there already :)

File: test/clj/rems/db/test_entitlements.clj [hunk 4]
```diff
@@ -127,131 +118,143 @@
...
+         {:status 200}
+         (fn [server]
+           (test-data/command! {:type :application.command/accept-licenses
+                                :application-id app-id
+                                :actor member
+                                :accepted-licenses [lic-id1 lic-id2]}) ; now accept all licenses
+
+           (entitlements/process-outbox!)
+
+           (testing "db"
```
[the remark this answers is not in the document]
Comment (opqdonut) [reply test/clj/rems/db/test_entitlements.clj]: good idea, done

**Pull request merged**